Sources/AngouriMath/Core/Parser.cs

```diff
@@ -47,6 +50,16 @@ public static Entity Parse(string source)
                 AngouriMathLexer.DefaultVocabulary.GetDisplayName(token.Type) is var type
                 && type is not PARENTHESIS_OPEN && type.EndsWith("('") ? FUNCTION_OPEN : type;
 
+            CommonToken GetPowerToken(AngouriMathLexer lexer, string tokenType)
```

Comment:
There's no reason to separate this into a new function, revert the pattern to using `(NUMBER or VARIABLE or PARENTHESIS_CLOSE, NUMBER)`

Reply:
Convert the if-statement to a ternary expression